test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -204,6 +203,37 @@ def test_zero_grad(self):
         self.assertFalse(m.weight.grad)
         self.assertFalse(m.bias.grad)
 
+    def test_constructor(self):
+        """Check the robustness of the ZeroRedundancyOptimizer constructor"""
+        self.dist_init(self.rank)
+
+        def assert_raises(exception_type, f):
+            """
+            Assert that the function `f` raises an exception of type
+            `exception_type`; redefines `assertRaises()` using try-except to
```

Comment:
I might miss sth. Could you please elaborate why `self.assertRaises` or `self.assertRaisesRegex` aren't sufficient here? 

https://github.com/pytorch/pytorch/blob/095cd6a0da255a5ed24d146898c53790f0e9c3f0/torch/testing/_internal/distributed/rpc/rpc_test.py#L603

Reply:
My mistake. I did not realize the correct usage of `self.assertRaises()` is as a context manager. I will use `self.assertRaises()`. Thanks for the tip!

Reply:
Fixed.